include/xively_types.h

```diff
@@ -148,6 +149,92 @@ typedef void( xi_user_subscription_callback_t )( xi_context_handle_t in_context_
                                                  xi_state_t state,
                                                  void* user_data );
 
+/**
+ * @name  xi_sft_on_file_downloaded_callback_t
+ * @brief At a the end of a Secure File Transfer (SFT) HTTP file download the application
+ * should report back the result to the Xively C Client through such typed function.
+ *
+ * The application calls this function reporting the result of the SFT file HTTP download.
+ * This function is the trigger for continuing the update package download inside the
+ * Xively C Client. The application receives this function pointer during a request for
+ * HTTP download, in the application callback (`xi_sft_url_handler_callback_t` see below).
+ *
+ * @param [in] callback_data the data pointer received as attribute `callback_data` before
+ * in the call of HTTP downloader function `xi_sft_url_handler_callback_t`. The
+ * application should not do anything with this pointer *just* pass it here back.
+ * @param [in] filename the name of the file got downloaded and stored on the non-volatile
+ * storage of the device
+ * @param flag_download_finished_successfully the result of the HTTP download: 1 - if
+ * file download finished successfully, 0 - if file download failed. In latter case the
+ * Xively C Client will try to fallback to MQTT file download.
+ */
+typedef void( xi_sft_on_file_downloaded_callback_t )(
+    void* callback_data,
+    const char* filename,
+    uint8_t flag_download_finished_successfully );
+
+/**
+ * @name  xi_sft_url_handler_callback_t
+ * @brief application entry point type for HTTP download of update package files
+ *
+ * The application has to define and pass the Xively C Client such typed function if
```

Comment:
The application much define and pass a pointer to to a function with this signature.  This function will be invoked by the Xively C Client if the SFT package contains resources too large to download over MQTT. Instead these resources will be served over an HTTP downloader service....

Reply:
rephrased but changed. Rephrased since the text above is not true. The function is called for EACH file if the function pointer is provided. And actually HTTP downloader service... hm, I don't know. Is it a service? We shouldn't touch this topic, we don't know what URL did the operator paste there. Yes, the broker currently accepts only HTTP, but in theory FTP is acceptable too. Client is unaware of the protocol.